### PR TITLE
Add a proc to check if a weather type is affecting an area

### PR DESCRIPTION
**Pull request by optimumtact**
A guy in discord wants this for his project

### CODE REVIEW

File: code/controllers/subsystem/weather.dm [hunk 1]
```diff
@@ -41,5 +41,14 @@ SUBSYSTEM_DEF(weather)
 		if(W.name == weather_name && W.target_z == Z)
 			W.telegraph()
 
+/datum/controller/subsystem/weather/proc/get_weather_affecting_area(area/A, var/weather_datum_type)
```
Comment (lzimann) [inline code/controllers/subsystem/weather.dm]: no `var/`

File: code/controllers/subsystem/weather.dm [hunk 2]
```diff
@@ -41,5 +41,14 @@ SUBSYSTEM_DEF(weather)
...
+	for(var/V in processing)
+		var/datum/weather/W = V
+		if(W.type != weather_datum_type)
```
Comment (lzimann) [inline code/controllers/subsystem/weather.dm]: why not `istype`

**Comment by praisenarsie** [comment]
interesting guy

File: code/controllers/subsystem/weather.dm (hunk 2 above)
Comment (lzimann) [reply code/controllers/subsystem/weather.dm]: or even an arg to check for subtypes or not

**Comment by epochayur** [comment]
are we getting snow days?

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**